src/devices/Rangefinder2DWrapper/Rangefinder2DWrapperROS.cpp

```diff
@@ -0,0 +1,315 @@
+
+    return true;
+}
+
+bool Rangefinder2DWrapperROS::initialize_ROS()
+{
+    rosNode = new yarp::os::Node(rosNodeName);   // add a ROS node
+    if (rosNode == nullptr)
+    {
+        yCError(RANGEFINDER2DWRAPPERROS) << " opening " << rosNodeName << " Node, check your yarp-ROS network configuration\n";
+        return false;
+    }
+    if (!rosPublisherPort.topic(rosTopicName))
+    {
+        yCError(RANGEFINDER2DWRAPPERROS) << " opening " << rosTopicName << " Topic, check your yarp-ROS network configuration\n";
+        return false;
+    }
+    return true;
+}
+
+/**
+  * Specify which sensor this thread has to read from.
+  */
+
+bool Rangefinder2DWrapperROS::attachAll(const PolyDriverList &device2attach)
+{
+    if (device2attach.size() != 1)
+    {
+        yCError(RANGEFINDER2DWRAPPERROS, "Rangefinder2DWrapper: cannot attach more than one device");
+        return false;
+    }
+
+    yarp::dev::PolyDriver * Idevice2attach = device2attach[0]->poly;
+
+    if (Idevice2attach->isValid())
+    {
+        Idevice2attach->view(sens_p);
+        Idevice2attach->view(iTimed);
+    }
+
+    if (nullptr == sens_p)
+    {
+        yCError(RANGEFINDER2DWRAPPERROS, "Rangefinder2DWrapper: subdevice passed to attach method is invalid");
+        return false;
+    }
+    attach(sens_p);
+
+    if(!sens_p->getDistanceRange(minDistance, maxDistance))
+    {
+        yCError(RANGEFINDER2DWRAPPERROS) << "Laser device does not provide min & max distance range.";
+        return false;
+    }
+
+    if(!sens_p->getScanLimits(minAngle, maxAngle))
+    {
+        yCError(RANGEFINDER2DWRAPPERROS) << "Laser device does not provide min & max angle scan range.";
+        return false;
+    }
+
+    if (!sens_p->getHorizontalResolution(resolution))
+    {
+        yCError(RANGEFINDER2DWRAPPERROS) << "Laser device does not provide horizontal resolution ";
+        return false;
+    }
+
+    PeriodicThread::setPeriod(_period);
+    return PeriodicThread::start();
+}
+
+bool Rangefinder2DWrapperROS::detachAll()
+{
+    if (PeriodicThread::isRunning())
+    {
+        PeriodicThread::stop();
+    }
+    sens_p = nullptr;
+    return true;
+}
+
+void Rangefinder2DWrapperROS::attach(yarp::dev::IRangefinder2D *s)
+{
+    sens_p = s;
+}
+
+void Rangefinder2DWrapperROS::detach()
+{
+    if (PeriodicThread::isRunning())
+    {
+        PeriodicThread::stop();
+    }
+    sens_p = nullptr;
+}
+
+bool Rangefinder2DWrapperROS::threadInit()
+{
+    return true;
+}
+
+bool Rangefinder2DWrapperROS::open(yarp::os::Searchable &config)
+{
+    Property params;
+    params.fromString(config.toString());
+
+    if (!config.check("period"))
+    {
+        yCError(RANGEFINDER2DWRAPPERROS) << "Rangefinder2DWrapper: missing 'period' parameter. Check you configuration file\n";
+        return false;
+    }
+    else
+        _period = config.find("period").asInt32() / 1000.0;
+
+    checkROSParams(config);
+
+    // call ROS node/topic initialization, if needed
```

Comment:
Since the wrapper is only dedicated to ROS, the ROS node/topic initialization is always needed. I'd modify the comment.